### ISSUE TITLE
Don't use `number_field_elements_from_algebraics` when constructing polygons

### ISSUE THREAD

**Issue by videlec** [flatsurf/sage-flatsurf#42]
If we don't pick the right number field (which is here `QQ[cos(alpha)]`) we end up with crazy coefficients
```
sage: polygons.triangle(4, 7, 22)
Polygon: (0, 0), (1, 0), (-17799064867529390581110150423415604011650744387432283511/1953890909590891666616379906072125869921297870743792781142079*a^19 + 71409052551592174955151033469798378037604417076221852511/3907781819181783333232759812144251739842595741487585562284158*a^18 + 190052921343937332201172045553860427173260259828749192933/217098989954543518512931100674680652213477541193754753460231*a^17 - 4948195845864956510738219441111494503105831955972361215630/1953890909590891666616379906072125869921297870743792781142079*a^16 - 59890035869003861986653374580550816662013935534393311401979/1953890909590891666616379906072125869921297870743792781142079*a^15 + 150456088221238765976031438375256575936377280774882914672831/1302593939727261111077586604048083913280865247162528520761386*a^14 + 895158398123285102173727174724820663527722822401947373944476/1953890909590891666616379906072125869921297870743792781142079*a^13 - 4512063935853680696953248103971998101372611496102166548541234/1953890909590891666616379906072125869921297870743792781142079*a^12 - 496727826295096723099318289240444950881660633563492969692037/217098989954543518512931100674680652213477541193754753460231*a^11 + 83758417350399139180381633563774696371436200770299788377702633/3907781819181783333232759812144251739842595741487585562284158*a^10 - 14758281744370911830226257037597260963639291863103903974983113/1953890909590891666616379906072125869921297870743792781142079*a^9 - 431466507047932824904675683576993870968723073520535522183698/4971732594378859202586208412397266844583455141841711911303*a^8 + 18618474579222203627520045956534013297561890244659310256746435/217098989954543518512931100674680652213477541193754753460231*a^7 + 189105947415578357827492522065943607501386351944657569260373555/1302593939727261111077586604048083913280865247162528520761386*a^6 - 135107128386523062547243532951812811375821549032552405156911954/651296969863630555538793302024041956640432623581264260380693*a^5 - 31845984690000626381120939076811547677617590128398165742880101/434197979909087037025862201349361304426955082387509506920462*a^4 + 111072413320347307581355619994870265017511876154387312733900292/651296969863630555538793302024041956640432623581264260380693*a^3 - 13613149249016747754582871421819717826378325956795358811452807/651296969863630555538793302024041956640432623581264260380693*a^2 - 51665050594377615423292031038742256583814248114972511800486136/1953890909590891666616379906072125869921297870743792781142079*a + 22009450854277521290800912902543851247822598990545082175070931/3907781819181783333232759812144251739842595741487585562284158, -50576523814248575105946557363934163543423299659239272889/3907781819181783333232759812144251739842595741487585562284158*a^19 + 83187752752640695069027996359879130033209418160467103803/3907781819181783333232759812144251739842595741487585562284158*a^18 + 807461006951315958101717739613169085886237167188731058156/651296969863630555538793302024041956640432623581264260380693*a^17 - 6167385045079250840818808380000502205856710662322602188265/1953890909590891666616379906072125869921297870743792781142079*a^16 - 170305563797408979645992611486603366985017773926280459367805/3907781819181783333232759812144251739842595741487585562284158*a^15 + 32114425465807702754341454208109056145818202797105339375535/217098989954543518512931100674680652213477541193754753460231*a^14 + 2594098589429733210509782163010918296750930071743615285622361/3907781819181783333232759812144251739842595741487585562284158*a^13 - 5853112987182818334016509869429635961416169120025561065342971/1953890909590891666616379906072125869921297870743792781142079*a^12 - 4723593311203573280285238618891592214986519198081179139026127/1302593939727261111077586604048083913280865247162528520761386*a^11 + 109280896035803744046983182810980603772672930732007754216633047/3907781819181783333232759812144251739842595741487585562284158*a^10 - 25827887853595516992774997010169921178096633901215414929330707/3907781819181783333232759812144251739842595741487585562284158*a^9 - 1125697193104431188308989518372120223074191817787162667515881/9943465188757718405172416824794533689166910283683423822606*a^8 + 201947097020808917424085695947165458502138656153208264445164402/1953890909590891666616379906072125869921297870743792781142079*a^7 + 733897826764878904007153782987671027843047921209965890259281249/3907781819181783333232759812144251739842595741487585562284158*a^6 - 56389795771201495008418006068849965881832218560290846990423046/217098989954543518512931100674680652213477541193754753460231*a^5 - 354762426967079228794413320305202289778492614795839758618366999/3907781819181783333232759812144251739842595741487585562284158*a^4 + 424364450379597951293591738998911862903897535131841869382014061/1953890909590891666616379906072125869921297870743792781142079*a^3 - 20120210137274809527943417963683825630658584496825389052598224/651296969863630555538793302024041956640432623581264260380693*a^2 - 14805196420708470003258321610898038389243293017463643169833597/434197979909087037025862201349361304426955082387509506920462*a + 4657007025905086430559957334069242258770033418680006559908222/651296969863630555538793302024041956640432623581264260380693)
```
The base ring picked up by `number_field_elements_from_algebraics` here is
```
Number Field in a with defining polynomial y^20 - 2*y^19 - 95*y^18 + 278*y^17 + 3261*y^16 - 12616*y^15 - 46510*y^14 + 248802*y^13 + 185375*y^12 - 2238022*y^11 + 1384857*y^10 + 8329244*y^9 - 11490204*y^8 - 10590074*y^7 + 25773561*y^6 - 2291586*y^5 - 19289743*y^4 + 10059654*y^3 + 1437271*y^2 - 1747768*y + 238117 with a = -0.4307429799237059?
```
The underlying problem is that `sin(2 pi / 66)` has a crazy minimal polynomial
```
sage: QQbar.zeta(66).imag().minpoly()
x^20 - 19/4*x^18 + 19/2*x^16 - 333/32*x^14 + 871/128*x^12 - 1391/512*x^10 + 2665/4096*x^8 - 1443/16384*x^6 + 195/32768*x^4 - 5/32768*x^2 + 1/1048576
```
So in the end this might just be unavoidable unless we use in a smarter way "real part" of complex number fields (see https://github.com/videlec/e-antic/issues/90).

**Comment by slel** [comment]
Working with `2*cos(pi/n)` should be nicer than `cos(pi/n)`.

See also

- Pierre-Vincent Koseleff, Fabrice Rouillier, Cuong Tran.
  On the Sign of a Trigonometric Expression.

**Comment by videlec** [comment]
(sort of) fixed via #44 and  #46
